src/hooks/useActiveDescendantListFocus/index.js

```diff
@@ -91,27 +104,33 @@ function useActiveDescendantListFocus({
     setVisualFocusItemIndex
   });
 
-  useCleanVisualFocusOnBlur({ focusedElementRef, visualFocusItemIndex, setVisualFocusItemIndex });
+  useCleanVisualFocusOnBlur({
+    focusedElementRef,
+    visualFocusItemIndex,
+    setVisualFocusItemIndex,
+    triggeredByKeyboard,
+    setTriggeredByKeyboard
+  });
 
   // this callback function is not needed anymore (the developer does not need to replace  the element's on click with this callback).
   // we keep it for backward compatibility
   const backwardCompatibilityCreateOnClickCallback = useCallback(
     itemIndex => event => onItemClick(event, itemIndex),
     [onItemClick]
   );
-  console.log(triggeredByKeyboard);
   return {
-    visualFocusItemIndex: triggeredByKeyboard.current ? visualFocusItemIndex : undefined,
-    visualFocusItemId: triggeredByKeyboard.current ? visualFocusItemId : undefined,
+    visualFocusItemIndex: triggeredByKeyboard ? visualFocusItemIndex : undefined,
```

Comment:
triggeredByKeyboard changed to be state instead of ref because we needed to trigger rerender